megablocks/layers/dmoe.py

```diff
@@ -242,27 +241,48 @@ def grouped_permute_and_compute(
             expert_capactiy,  # unused
             top_k):
 
-        # Route the tokens for MoE computation.
+        # Route the tokens for MoE MLP computation.
         x = x.view(-1, x.shape[-1])
-        x = ops.gather(
-            x,
-            indices,
-            bin_ids,
-            bins,
-            top_k)
-
-        # Perform the expert computation.
-        x = self.mlp(x, tokens_per_expert)
+        if self.args.fp8:
+            padded_tokens_per_expert = ops.round_up(tokens_per_expert, self.blocking)
```

Comment:
You should only pad to the TE requirement (8, IIRC) rather than to `blocking`, which is 128.

Reply:
fixed